pkg/argocd/update.go

```diff
@@ -551,13 +551,24 @@ func mergeHelmOverride(t *helmOverride, o *helmOverride) {
 }
 
 func mergeKustomizeOverride(t *kustomizeOverride, o *kustomizeOverride) {
-	for _, image := range *o.Kustomize.Images {
-		idx := t.Kustomize.Images.Find(image)
-		if idx != -1 {
-			(*t.Kustomize.Images)[idx] = image
-			continue
+	for _, newImage := range *o.Kustomize.Images {
+		found := false
+		newContainerImage := image.NewFromIdentifier(string(newImage))
+		for idx, existingImage := range *t.Kustomize.Images {
+			existingContainerImage := image.NewFromIdentifier(string(existingImage))
+			if newContainerImage.ImageName == existingContainerImage.ImageName &&
+				newContainerImage.RegistryURL == existingContainerImage.RegistryURL {
```

Comment:
From the tests that I performed, RegistryURL is nil all the time.  Is there another test that will test this condition with non-nil values?

Reply:
`RegistryURL` has a default value "" and should not be nil.

The first 4 sub-tests don't use any registry url, so they should be able to test this case.